ios/Classes/SwiftFlutterWeatherSensorPlugin.swift

```diff
@@ -1,14 +1,71 @@
 import Flutter
 import UIKit
+import Combine
 
-public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin {
+@available(iOS 13.0, *)
+public class SwiftFlutterWeatherSensorPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+  private var eventSink: FlutterEventSink?
+  private var sinkSubscription: AnyCancellable?
+    
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let methodChannel = FlutterMethodChannel(name: "flutter_weather_sensor_plugin", binaryMessenger: registrar.messenger())
+    let eventChannel = FlutterEventChannel(name: "flutter_weather_sensor_plugin/readings", binaryMessenger: registrar.messenger())
+      
     let instance = SwiftFlutterWeatherSensorPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addMethodCallDelegate(instance, channel: methodChannel)
+    eventChannel.setStreamHandler(instance)
   }
 
   public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
+    switch call.method {
+      case "startSensorReadings":
+        startSensorReadings()
+        result("success")
+      case "stopSensorReadings":
+        stopSensorReadings()
+        result("success")
+      case "set":
+        if let args = call.arguments as? Dictionary<String, Any>,
+          let interval = args["readingInterval"] as? UInt {
+            setInterval(interval)
+          } else {
+            result(FlutterError.init(code:"invalidArgs", message: "Reading interval must be a positive integer.", details: "Expected 1 UInt arg."))
+          }
+        result("success")
```

Comment:
result("success") looks like it is called in both the failure and success cases. I am guessing that is not correct. The code in this case would be better as a Swift guard:
```
guard 
    let args = call.arguments as? Dictionary<String, Any>,
    let interval = args["readingInterval"] as? UInt
else {
    result(FlutterError.init(code:"invalidArgs", message: "Reading interval must be a positive integer.", details: "Expected 1 UInt arg."))
    return
}

setInterval(interval)
result("success")
```

Also I suppose the "success" string could be defined so as not to be a "magic string"